tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs

```diff
@@ -7615,12 +7634,16 @@ private async Task<TResult> ExecuteWorkerAsync<TWorkflow, TResult>(
         TemporalWorkerOptions? options = null,
         IWorkerClient? client = null)
     {
-        options ??= new();
-        options = (TemporalWorkerOptions)options.Clone();
+        options = PrepareOptions<TWorkflow>((TemporalWorkerOptions?)options?.Clone() ?? new());
+        using var worker = new TemporalWorker(client ?? Client, options);
+        return await worker.ExecuteAsync(() => action(worker));
+    }
+
+    private static TemporalWorkerOptions PrepareOptions<TWorkflow>(TemporalWorkerOptions options)
```

Comment:
```suggestion
    private static TemporalWorkerOptions PrepareWorkerOptions<TWorkflow>(TemporalWorkerOptions options)
```

Pedantic, don't have to change